Add a recipe for uxntal-mode.

non opened the pull request:
### Brief summary of what the package does

This major mode supports programming in the Unxtal assembly language. It
includes syntax highlighting as well as support for compiling, navigation, etc.

The language is documented at https://wiki.xxiivv.com/site/uxntal.html.

### Direct link to the package repository

https://github.com/non/tal-mode

### Your association with the package

I am the creator and maintainer of tal-mode

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
I can point out an existing tal-mode here which we may want to avoid a collision with:
- tal-mode: https://github.com/emacsmirror/tal-mode.git (via emacsmirror)

In the meantime, quick first pass --

`tal-mode.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- tal-mode.el#L276: You can use `not` or `null`
- tal-mode.el#L306: You can use `not` or `null`
- tal-mode.el#L194: Prefer `define-derived-mode` rather than manually setting the mode up
- tal-mode.el#L206 and 213: Loading a package should rarely add hooks (defining a hook shouldn't be necessary to run this code, should it?  Although I may be missing something)
- ```

Package and license:
- tal-mode.el's license line is informal or difficult to detect -- consider official license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/)

non commented:
@riscy Thanks for your feedback, I'm working through it now.

non commented:
@riscy Thanks for your feedback. I think I addressed most of the issues you found.

I have one remaining question: your script discourages me from adding hooks. However, I can't figure out any other way to specify the right `M-x compile` behavior for my mode (it's unlikely that `make` is a command the user would expect to run).

I see a few ways forward:
1. Leave it as-is.
2. Remove the hook, but leave the function around to make easy for users to set this up themselves.
3. Configure the `compile-command` some other way.
4. Accomplish the same task (assembling the current buffer) some other way.

Do you have any recommendations?

riscy commented:
> your script discourages me from adding hooks. However, I can't figure out any other way to specify the right M-x compile behavior for my mode

I may be missing something, but since you're adding a hook to `uxntal-mode-hook`, couldn't you just call `(uxntal-assemble-buffer)` directly when the user activates `uxntal-mode`?

non commented:
@riscy Aha, I see what you mean now!

Done: https://github.com/non/uxntal-mode/commit/a72ad9c329ea3b810a58a47d3d0f018247aa317a

riscy commented:
Thanks, looks good!  It looks like this comment still applies:

- tal-mode.el's license line is informal or difficult to detect -- consider official license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/)

and it's an easy fix, so I won't block on this.  Merging. :)  And, welcome to MELPA!
